cli/JSXV4.md

```diff
@@ -373,3 +373,36 @@ let p: A.props<_> = {x: "x", y: "y"}
 <A x="X" {...p}>
 <A {...p} {...p1}>
 ```
+
+### Shared props type (new feature)
+
+V4 introduces support to set a props type from the outside. It allows sharing of the props type between components.
+
+```rescript
+type sharedProps<'a> = {x: 'a, y: string}
+
+module A = {
+  @react.component(:sharedProps<'a>)
```

Comment:
Would you clarify in the text above what's the role of `<'a>`. Does the name `'a` matter, or is it only used to specify the number of type arguments?

Reply:
Not sure I understand: or can be 1 zero type args, or 2 more than zero. And the user needs to specify which one is true?

Reply:
Sorry for making communication cost. My point was that I expected there were some use cases:
```rescript
type sp1 = ...

@react.component(:sp1)
```
```rescript
type sp2<'a> = ...

@react.component(:sp2<'a>)
// or
@react.component(:sp2<string>)
```
```rescript
type sp3<'a, 'b, ...> = ...

@react.component(:sp3<'a, 'b, ...>)
// or
@react.component(:sp3<string, 'b, ...>)
```
The text is saying that the type argument(s) is available, the name of `'a` wouldn't matter.

Reply:
> Sorry for making communication cost. My point was that I expected there were some use cases:
> 
> ```rescript
> type sp1 = ...
> 
> @react.component(:sp1)
> ```
> 
>  ```rescript
> type sp2<'a> = ...
> 
> @react.component(:sp2<'a>)
> // or
> @react.component(:sp2<string>)
> ```
> 
>  ```rescript
> type sp3<'a, 'b, ...> = ...
> 
> @react.component(:sp3<'a, 'b, ...>)
> // or
> @react.component(:sp3<string, 'b, ...>)
> ```
> 
> The text is saying that the type argument(s) is available. The name of `'a` nor the number of type arguments wouldn't matter.

The name does not matter. Does the number matter?

Reply:
Does it work also if the type definition has zero type arguments?

Reply:
If you mean this case:
```rescript
type sp1 = ...

@react.component(:sp1)
```
Yes, it works. https://github.com/rescript-lang/syntax/blob/master/tests/ppx/react/expected/sharedProps.res.txt#L38

Reply:
I read the source code. So the number of type variables does matter. The name matters only so much as it normally matters, but not less.

So I think what's missing in the doc is one example with 0 and one 2 type variables.

Reply:
One should also remark on the fact that a type called `prop` cannot be defined in the current module or there will be a type error due to type redefinition. (It could at most be defined outside the current module)

Reply:
> One should also remark on the fact that a type called `props` cannot be defined in the current module or there will be a type error due to type redefinition. (It could at most be defined outside the current module)

I think this is already noted as https://github.com/rescript-lang/syntax/blob/master/cli/JSXV4.md#transformation-for-component-definition

Reply:
How does it look? https://github.com/rescript-lang/syntax/pull/701/commits/c052cb99bf5a0b99a247b929a3bf170635ca0dc8